lib/ansible/modules/cloud/amazon/kms_facts.py

```diff
@@ -0,0 +1,313 @@
+        type: str
+        returned: always
+        sample: arn:aws:iam::01234567890:root
+      key_id:
+        description: The key ARN to which the grant applies.
+        type: str
+        returned: always
+        sample: arn:aws:kms:ap-southeast-2:123456789012:key/abcd1234-abcd-1234-5678-ef1234567890
+      name:
+        description: The friendly name that identifies the grant
+        type: str
+        returned: always
+        sample: xyz
+      operations:
+        description: The list of operations permitted by the grant
+        type: list
+        returned: always
+        sample:
+          - Decrypt
+          - RetireGrant
+      retiring_principal:
+        description: The principal that can retire the grant
+        type: str
+        returned: always
+        sample: arn:aws:sts::0123456789012:assumed-role/lambda_xyz/xyz
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import AWSRetry, camel_dict_to_snake_dict, HAS_BOTO3
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+import traceback
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def get_kms_keys_with_backoff(connection):
+    paginator = connection.get_paginator('list_keys')
+    return paginator.paginate().build_full_result()
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def get_kms_tags_with_backoff(connection, key_id, **kwargs):
+    return connection.list_resource_tags(KeyId=key_id, **kwargs)
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def get_kms_grants_with_backoff(connection, key_id, **kwargs):
+    params = dict(KeyId=key_id)
+    if kwargs.get('tokens'):
+        params['GrantTokens'] = kwargs['tokens']
+    paginator = connection.get_paginator('list_grants')
+    return paginator.paginate(**params).build_full_result()
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def get_kms_metadata_with_backoff(connection, key_id):
+    return connection.describe_key(KeyId=key_id)
+
+
+def get_kms_tags(connection, module, key_id):
+    # Handle pagination here as list_resource_tags does not have
+    # a paginator
+    kwargs = {}
+    tags = []
+    more = True
+    while more:
+        try:
+            tag_response = get_kms_tags_with_backoff(connection, key_id, **kwargs)
+            tags.extend(tag_response['Tags'])
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg="Failed to obtain key tags",
+                             exception=traceback.format_exc(),
+                             **camel_dict_to_snake_dict(e.response))
+        if tag_response.get('NextMarker'):
+            kwargs['Marker'] = tag_response['NextMarker']
+        else:
+            more = False
+    return boto3_tag_list_to_ansible_dict(tags)
```

Comment:
I think this should be `return boto3_tag_list_to_ansible_dict(tags, 'TagKey', 'TagValue')` to get any tags back
http://boto3.readthedocs.io/en/latest/reference/services/kms.html#list_resource_tags